### DISCUSSION TITLE
Can someone explain, is nesting pages possible?

### DISCUSSION THREAD

**Discussion by nogginfuel** [datenstrom/community#159] [Ask a question] [answered]
Earlier I was trying to add nested pages, but It didn't seem to be working for me. I was trying to accomplish something like the following:
**\- 1-shop**
_--- 1-productA
--- 2-productB
--- 3-productC_

Adding sub folders didn't seem to work very well. The only other way I can think about doing this is setting up the shop area like a blog, but I'm not sure if that would work. Any help or feedback is greatly appreciate

**Comment by BsNoSi** [comment]
Do you want to have a menu with sub entries or is it a challenge of organizing it?

I have realized this for a [shopping street in Frankfurt/Main, Germany](http://www.oederwegffm.de). If you click on `Angebote` in menu (top), you are led to a subpage, organizing the groups and this leads to a page with all pages within the group. On each page one shop is presented with a short description.

This can be solved as well with sidebars… .

**Comment by nogginfuel** [comment]
@BsNoSi Thanks for your input. I understand what you are trying to present with that example. It is similar to what I was looking for but I've attached an image to add clarity.

![nested](https://cloud.githubusercontent.com/assets/2807575/12983581/cf21a5ae-d0b7-11e5-9545-31ec7df11817.png)

So, I've been able to successfully add "subpages" under some of the folders as you see under the "Shop" folder. However, I have been unsuccessful in trying to add subpages to the folder that is seen as the "home" directory for the overall YellowCMS install. Meaning, if I add a subfolder under "Home" as seen the picture, the links will not properly navigate to the content. Also hand typing the until URL path into the address bar fires off a "Content Not Found" message instead of the file.

So circling back to my question from yesterday with additional clarity, Is it possible to have underlying pages underneath the home/primary directory for Yellow?

**Comment by markseuffert** [comment, marked as answer]
Yes, it's possible to have subfolders. 

Let's take a look from the viewpoint of the website URL. The home folder is what's available at `http://website/`. The shop folder is available at `http://website/shop/`. If you want URLs like `http://website/shop/producta/` and `http://website/shop/productb/`, then you create the corresponding subfolders in your [content folder](https://datenstrom.se/yellow/help/how-to-change-the-content). Not in your home folder. Here's an example:

![screen shot 2016-03-12 at 13 35 13](https://cloud.githubusercontent.com/assets/2429165/13722984/5110f3ba-e857-11e5-9d08-f9fecb7cb5fa.png)

You should be able to create the content structure you want. You can make it flat and put multiple files in a folder. You can make it nested and put files into subfolders, as shown in the screenshot above. If you do use multiple subfolders, you might want to add a custom navigation or breadcrumbs or jump page as @BsNoSi mentioned.

Edit: Updated screenshot

**Comment by nogginfuel** [comment]
Yes, thank you for the additional explanation and examples. I guess I wasn't thinking that the first folder would redirect to the regular domain which would make all the underlying pages not point / redirect properly. This has definitely helped. Thanks everyone who added to the discussion.

**Comment by markseuffert** [comment]
Good luck with your shop :smile:

**Comment by nogginfuel** [comment]
Thanks, I've got dueling banjos at the moment. Two different installs of Yellow CMS one trying Shoprocket.co e-com API and the other is going to use Snipcart.com hybrid e-com approach. I have to see which will better fit my needs. More to come...

**Comment by pftnhr** [comment]
Hello everyone, I don't want to start a new thread, so I'll put my question here:

I would like to output the content of several sub folders with something like `<?php echo $this->yellow->page->getPage("producta")->getContent() ?>` on one page, e.g. `shop.html`. I know that `getPage()` only works with shared pages, but I hope that I don't have to put different products in one level in the `shared` folder to keep some structure.

**Reply by annaesvensson** [reply]
Let's stick to one question per discussion... you know the drill. :smile:

**Reply by pftnhr** [reply]
Aye! 😄
